x/auth/genesis.go

```diff
@@ -12,9 +12,18 @@ import (
 // a genesis port script to the new fee collector account
 func InitGenesis(ctx sdk.Context, ak keeper.AccountKeeper, data types.GenesisState) {
 	ak.SetParams(ctx, data.Params)
-	data.Accounts = types.SanitizeGenesisAccounts(data.Accounts)
 
-	for _, a := range data.Accounts {
+	accounts := make([]types.GenesisAccount, len(data.Accounts))
+	for i, genAcc := range data.Accounts {
+		acc, ok := genAcc.GetCachedValue().(types.GenesisAccount)
```

Comment:
a test case would be nice here

Reply:
I should just use `types.ConvertAccountsAny` here instead and test `ConvertAccountsAny` separately in `x/auth/types/genesis_test.go` as you mentioned in your last comment.